src/models/ModelBuilder.cc

```diff
@@ -3,23 +3,136 @@
 //
 
 #include "ModelBuilder.h"
+#include <unordered_set>
 
-void ModelBuilder::processSubstitutions(Logic::SubstMap const & subst) {
-    Logic::SubstMap copy;
-    subst.copyTo(copy);
-    logic.substitutionsTransitiveClosure(copy);
-    auto assignCopy = this->assignment;
-    auto model = this->build();
-    for (auto key : copy.getKeys()) {
-        if (logic.isIte(key)) {
-            // We store only values of reals variables
+std::unique_ptr<Model> ModelBuilder::build() {
+    std::unordered_map<SymRef,TemplateFunction,SymRefHash> builtDefinitions;
+    for (auto & symbolSigVal : definitions) {
```

Comment:
Apparently it can't because we move the signature a few lines below for the constructor of `TemplateFunction`.  We could copy it instead of course, but I'm not sure if that's the right solution.  This also means that `ModelBuilder::build()` should be called only once since `definitions` will have undefined values after the first call.

Reply:
OK, I missed the `move`. It's fine this way.